pkg/latte/intent_change_password.go

```diff
@@ -0,0 +1,74 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentChangePassword{})
+}
+
+var IntentChangePasswordSchema = validation.NewSimpleSchema(`{}`)
+
+type IntentChangePassword struct {
+}
+
+func (*IntentChangePassword) Kind() string {
+	return "latte.IntentChangePassword"
+}
+
+func (*IntentChangePassword) JSONSchema() *validation.SimpleSchema {
+	return IntentChangePasswordSchema
+}
+
+func (*IntentChangePassword) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	if len(w.Nodes) == 0 {
+		return []workflow.Input{
+			&InputTakeLoginID{},
```

Comment:
Changing the password is a feature that requires an authenticated user, `IntentChangePassword` should not take the login id to search the user.

Instead, the user id should be resolved from the user session which can be obtained through the context. e.g. https://github.com/authgear/authgear-server/blob/c81d8eda8da7ceba70ccc853f25a3c692957abe3/pkg/latte/intent_verify_user.go#L44